test/kubernetes/deployment.go

```diff
@@ -0,0 +1,222 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetes
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	networking "k8s.io/api/networking/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/yaml"
+)
+
+// EchoService name of the deployment for the echo app
+const EchoService = "echo"
+
+// EchoContainer container image name
+//const EchoContainer = "gcr.io/k8s-staging-ingressconformance/echoserver@sha256:e4e132da40d303f4b50e183b977ed1eb5c6db6eb0ddbdaa19565d14d49e05940"
+const EchoContainer = "aledbf/echoserver"
+
+const deploymentTemplate = `
```

Comment:
Given the number of `%v` arguments, it might be easier to read if turned it into a go-template.

Reply:
done